agent/consul/leader.go

```diff
@@ -662,6 +662,9 @@ func (s *Server) initializeACLs(upgrade bool) error {
 			if s.IsACLReplicationEnabled() {
 				s.startLegacyACLReplication()
 			}
+			// return early as we don't want to start new ACL replication
+			// or ACL token reaping as these are new ACL features.
+			return nil
```

Comment:
How hard would it be to get a test that asserts on this? Seems like a potentially easy bug to reintroduce with even a small refactor.

Reply:
I added one. Usually multi-dc version dependent tests are hard but this one was ok as I could force this particular path by just not connecting to the primary DC